### PR TITLE
Bugfix and new simple example

### PR DESCRIPTION
**Pull request by microcosm**
Hi Jim,

I wanted to use ofxTriangle in a project and discovered a small bug in the OpenCv example code. I also found it a bit of a heavyweight example and made my own proof of concept before adding it to my other project.

I thought you might want to use the additions in the original branch :) - I also added a Readme.

Cheers
Andy

### CODE REVIEW

File: example-simple/testApp.cpp [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	glBegin(GL_TRIANGLES);
+	for (int i = 0; i < triangle.nTriangles;i++){
+		glTexCoord2f(triangle.triangles[i].a.x, triangle.triangles[i].a.y);
+		glVertex2f(triangle.triangles[i].a.x, triangle.triangles[i].a.y);
+        
+		glTexCoord2f(triangle.triangles[i].b.x, triangle.triangles[i].b.y);
+		glVertex2f(triangle.triangles[i].b.x, triangle.triangles[i].b.y);
+        
+		glTexCoord2f(triangle.triangles[i].c.x, triangle.triangles[i].c.y);
+		glVertex2f(triangle.triangles[i].c.x, triangle.triangles[i].c.y);
```
Comment (obviousjim) [inline example-simple/testApp.cpp]: could you switch all these calls to openFrameworks ofBeginShape() calls?

File: example-simple/testApp.cpp [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    
+    ofPoint point5;
+    point5.x = 500;
+    point5.y = 100;
+    point5.z = 0;
+    
+    ofPoint point6;
+    point6.x = 500;
+    point6.y = 400;
+    point6.z = 0;
```
Comment (obviousjim) [inline example-simple/testApp.cpp]: It'd be better style to use the point constructors here:
ofPoint point6(500,400,0)

File: example-simple/testApp.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#ifndef _TEST_APP
+#define _TEST_APP
+
```
Comment (obviousjim) [inline example-simple/testApp.h:3]: I think we've adopted using #pragma once, at least I have ;)

File: example-simple/testApp.h [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	void setup();
+	void update();
+	void draw();
+	void keyPressed(int key);
+
+private:
+	ofxTriangle             triangle;
+    
+    ofFbo buffer;
+    vector<ofPoint> points;
```
Comment (obviousjim) [inline example-simple/testApp.h]: Why not use ofPolyLine?

File: example-simple/testApp.cpp [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    
+    createPointsVector();
+}
+
+
+void testApp::update(){
+
+    //Draw anything you like inside the buffer
+    buffer.begin();
+    glClearColor(0, 0, 0, 1.0f);
```
Comment (obviousjim) [inline example-simple/testApp.cpp]: Best to use ofClear() here

File: example-simple/testApp.cpp [hunk 6]
```diff
@@ -0,0 +1,112 @@
+#include "testApp.h"
+
+void testApp::setup(){
+	
+	ofBackground(50, 50, 50);
+    ofSetFrameRate(60);
+    ofSetWindowShape(600, 500);
+	
+    buffer.allocate(640, 480, GL_RGBA, 1);
```
Comment (obviousjim) [inline example-simple/testApp.cpp]: no need to pass 1 as it's the default param

**Comment by obviousjim** [comment]
Hey this is great. I think it just needs a bit of love to make it less direct OpenGL and more openFrameworks-ified. 

Maybe there is no need for an FBO too? I don't see what it adds to the example as far as learning how to use ofxTriangle

File: example-simple/testApp.h (hunk 3 above)
Comment (microcosm) [inline example-simple/testApp.h:3]: I read that ifndef is portable where pragma is not: http://www.cplusplus.com/forum/beginner/7877/

However I have no strong opinions, whatever is in line with the community.

(Nb you have used ifndef in the existing testApp.h in this repo, I can change that too)

File: example-simple/testApp.cpp (hunk 6 above)
Comment (microcosm) [inline example-simple/testApp.cpp]: done

File: example-simple/testApp.cpp (hunk 5 above)
Comment (microcosm) [inline example-simple/testApp.cpp]: done

File: example-simple/testApp.h (hunk 4 above)
Comment (microcosm) [inline example-simple/testApp.h]: done

File: example-simple/testApp.cpp (hunk 2 above)
Comment (microcosm) [inline example-simple/testApp.cpp]: done

File: example-simple/testApp.cpp (hunk 1 above)
Comment (microcosm) [inline example-simple/testApp.cpp]: see below

**Comment by microcosm** [comment]
I think you're right that FBO adds a layer of complication, which in a simple example may be wrong. However as a noob I found it useful to think that you can do more with a shape than just fill it. I've made it into 2 examples now and I think it's a lot clearer, let me know what you think.

**Comment by obviousjim** [comment]
hey this is great!

check out this commit:
https://github.com/obviousjim/ofxTriangle/commit/956143329a016d7b03706c847ea4f9ba15afd29b

The libs folder needed some reworking to be compatible with standard addon
strategy and to work with the project generator. I went ahead and added
xcode projects because they are just really nice to have around

I think we should update this lib to use ofMesh internally like i've done
for ofxDelaunay
https://github.com/obviousjim/ofxDelaunay

On Wed, Aug 29, 2012 at 12:57 AM, Andrew McWilliams <
notifications@github.com> wrote:

> I think you're right that FBO adds a layer of complication, which in a
> simple example may be wrong. However as a noob I found it useful to think
> that you can do more with a shape than just fill it. I've made it into 2
> examples now and I think it's a lot clearer, let me know what you think.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/obviousjim/ofxTriangle/pull/1#issuecomment-8096714.

## 
- James

**Pull request merged**